static/app/views/performance/newTraceDetails/traceDrawer/details/span/sections/keys.tsx

```diff
@@ -64,6 +76,93 @@ function partitionSizes(data: RawSpanType['data']): {
   };
 }
 
+function getSpanTimeWindow(timestamp: number) {
+  // Convert 30 minutes to miliseconds
+  const thirtyMinutesInMiliSeconds = 30 * 60 * 1000;
+
+  // Calculate the start and end of the time window
+  const startTime = timestamp - thirtyMinutesInMiliSeconds;
+  const endTime = timestamp + thirtyMinutesInMiliSeconds;
+
+  return {
+    startTime: startTime,
+    endTime: endTime,
+  };
+}
+
+function getNonSizeKeyActionButtonData({
+  organization,
+  extractionRules,
+  spanAttribute,
+  projectId,
+  spanTimestamp,
+}: {
+  organization: Organization;
+  spanAttribute: string;
+  spanTimestamp: number;
+  extractionRules?: MetricsExtractionRule[];
+  projectId?: string;
+}): Pick<KeyValueListDataItem, 'actionButton' | 'actionButtonAlwaysVisible'> {
+  if (!hasCustomMetricsExtractionRules(organization)) {
+    return {};
+  }
+
+  const extractionRule = extractionRules?.find(
+    rule => rule.spanAttribute === spanAttribute
+  );
+
+  if (extractionRule) {
+    const virtualMRI = createVirtualMRI(extractionRule);
+    const spanTimeWindow = getSpanTimeWindow(spanTimestamp);
+    return {
+      actionButton: (
+        <LinkButton
+          size="xs"
+          to={getMetricsUrl(organization.slug, {
+            start: getUtcDateString(spanTimeWindow.startTime),
+            end: getUtcDateString(spanTimeWindow.endTime),
+            project: [extractionRule.projectId],
+            widgets: [
+              {
+                mri: virtualMRI,
+                displayType: MetricDisplayType.BAR,
+                aggregation: extractionRule.aggregates[0],
+                query: '',
+                groupBy: undefined,
+                condition: extractionRule.conditions[0].id,
+              },
+            ],
+          })}
+        >
+          {t('Open in Metrics')}
```

Comment:
We used to have this as a blue link. It's way less heavy that way. See "View Similar Spans" button in your video. 👍

Reply:
+1, the links in these section are all inline and we avoid the button as there is already a lot going on in this entire UI here :)